### DISCUSSION TITLE
Roborock S7 cleaning details missing?

### DISCUSSION THREAD

**Discussion by shred86** [rytilahti/python-miio#1208] [General]
First off, thank you @rytilahti for creating this awesome library! I recently purchased a Roborock S7 and have it integrated through Home Assistant using the Xiaomi Miio integration. I noticed there’s no entity that shows the [cleaning details](https://github.com/rytilahti/python-miio/blob/f4437106741b9680cc7f67f17200a71094630cdd/miio/integrations/vacuum/roborock/vacuumcontainers.py#L270), but rather it seems to only show the [cleaning summary](https://github.com/rytilahti/python-miio/blob/f4437106741b9680cc7f67f17200a71094630cdd/miio/integrations/vacuum/roborock/vacuumcontainers.py#L216) information. Specifically, I’m looking to have the end date/time of a cleaning available in Home Assistant to set up some automations.

I asked this question on Reddit and some users mention the cleaning details are available in Home Assistant with other Roborock models. I noticed in the Mi Home app the Roborock S7 does have a “Cleaning History” section with the start date/time, total area and total time, so it seems like the information is there. Does anyone know if it’s just not possible to have that information exposed through this library with the Roborock S7?

**Comment by rytilahti** [comment]
You're welcome! Did you take a look at the device page? The information about last cleaning should be available on all models, but it might be disabled per default.

**Reply by shred86** [reply]
Yep, I did. The Roborock S7 had 3 entities enabled and 10 entities disabled by default, but the last cleaning data seems to be missing. These are all the entities available (first three were enabled by default):

- binary_sensor.roborock_s7_mop_attached
- binary_sensor.roborock_s7_water_box_attached
- binary_sensor.roborock_s7_water_shortage
- sensor.roborock_s7_dnd_end
- sensor.roborock_s7_dnd_start
- sensor.roborock_s7_filter_left
- sensor.roborock_s7_main_brush_left
- sensor.roborock_s7_sensor_dirty_left
- sensor.roborock_s7_side_brush_left
- sensor.roborock_s7_total_clean_area
- sensor.roborock_s7_total_clean_count
- sensor.roborock_s7_total_duration
- sensor.roborock_s7_total_dust_collection_count

**Reply by shred86** [reply]
Weird. I enabled all of the disabled entities and suddenly more are showing up. I now have 18 total entities which includes everything I was missing!
